src/form-builder/components/FormBuilder.jsx

```diff
@@ -85,75 +151,165 @@ export default class FormBuilder extends Component {
-              const formResource = {
-                form: {
-                  name: formName,
-                  uuid: response.uuid,
-                },
-                value: JSON.stringify(value),
-                uuid: '',
-              };
-              self.props.saveFormResource(formResource, translations);
-            })
-              .catch(() => {
-                const formUuid = self.getFormUuid(formName);
-                value.uuid = formUuid;
-                const params =
-                  'v=custom:(id,uuid,name,version,published,auditInfo,' +
-                  'resources:(value,dataType,uuid))';
-                httpInterceptor
-                  .get(`${formBuilderConstants.formUrl}/${formUuid}?${params}`)
-                  .then((data) => {
-                    const formResource = {
-                      form: {
-                        name: formName,
-                        uuid: formUuid,
-                      },
-                      value: JSON.stringify(value),
-                      uuid: data.resources[0].uuid,
-                    };
-
-                    self.props.saveFormResource(formResource, translations);
-                  });
-              });
-          });
+        const formData = JSON.parse(reader.result);
+        const formsValidationPromises = [];
+        const validateFormJsonPromise = self.validateFormJson(fileName, formData);
+        if (validateFormJsonPromise !== null) {
+          formsValidationPromises.push(validateFormJsonPromise);
         }
+        self.import(formsValidationPromises);
       } catch (error) {
-        self.props.onValidationError('Error Importing.. Please import a valid form');
+        self.updateImportErrors(fileName, self.parseErrorMessage);
+        self.downloadErrorsFile(self.importErrors);
       }
     };
-
     reader.readAsText(file[0]);
   }
 
+  validateFormJson(fileName, formData) {
+    const self = this;
+    const { formJson, translations } = formData;
+    const formName = formJson.name;
+    const value = JSON.parse(formJson.resources[0].value);
+    const form = {
+      name: formName,
+      version: '1',
+      published: false,
+    };
+    if (!formHelper.validateFormName(formName)) {
+      self.updateImportErrors(fileName, 'Invalid File Name');
+    } else {
+      const conceptValidationPromise = self.fixuuid(value, fileName);
+      return conceptValidationPromise.then(() => {
+        if (self.formConceptValidationResults[fileName]) {
+          const validationResults = self.formConceptValidationResults[fileName];
+          let message = 'Concept validation error: \n';
+          validationResults.forEach(v => {
+            message = message.concat(`${v}\n`);
+          });
+          self.updateImportErrors(fileName, message);
+        } else {
+          self.formJSONs.push({ form, value, formName, translations });
+        }
+      });
+    }
+    return null;
+  }
+
+  import(formsValidationPromises) {
```

Comment:
either we can club both import and importForms or give a proper method names for both of them as its hard to figure it out the diff from method names